src/main.cc

```diff
@@ -1484,6 +1486,8 @@ RegisterModules()
 int
 SquidMain(int argc, char **argv)
 {
+    ConfigFile = xstrdup(DEFAULT_CONFIG_FILE);
+
```

Comment:
This change is what produces `squid -v` output:
```
Squid Cache: Version 7.0.0-VCS
Service Name: squid
Running Configuration in: /usr/local/squid/etc/squid.conf
Build configure options: ....
```

instead of:
```
Squid Cache: Version 7.0.0-VCS
Service Name: squid
Running Configuration in: (null)
Build configure options: ....
```

Reply:
> This change is what produces `squid -v` output: ...

Yes, I am aware. This fact does not address the concern detailed in this change request. Clearly, the same output[^1] can be produced without this problematic code change.

[^1]: The problems with that output are discussed [elsewhere](https://github.com/squid-cache/squid/pull/1790/files#r1576166997). This change request is not about those problems.